packages/pigeon/lib/dart_generator.dart

```diff
@@ -8,13 +8,45 @@ import 'generator_tools.dart';
 /// Options that control how Dart code will be generated.
 class DartOptions {
   /// Constructor for DartOptions.
-  DartOptions({this.isNullSafe = true, this.copyrightHeader});
+  const DartOptions({this.isNullSafe = true, this.copyrightHeader});
 
   /// Determines if the generated code has null safety annotations (Dart >=2.12 required).
-  bool isNullSafe;
+  final bool isNullSafe;
 
   /// A copyright header that will get prepended to generated code.
-  Iterable<String>? copyrightHeader;
+  final Iterable<String>? copyrightHeader;
+
+  /// Creates a [DartOptions] from a Map representation where:
+  /// `x = DartOptions.fromMap(x.toMap())`.
+  static DartOptions fromMap(Map<String, Object> map) {
+    return DartOptions(
+      isNullSafe:
+          // ignore: avoid_bool_literals_in_conditional_expressions
+          map.containsKey('isNullSafe') ? (map['isNullSafe'] as bool?)! : true,
+      copyrightHeader: map.containsKey('copyrightHeader')
+          ? map['copyrightHeader'] as Iterable<String>?
+          : null,
+    );
+  }
+
+  /// Converts a [DartOptions] to a Map representation where:
+  /// `x = DartOptions.fromMap(x.toMap())`.
```

Comment:
Nit: I had to read this expression a couple of times. Maybe just say "...a Map representation that can be parsed with `fromMap`"?

Reply:
I bet it's the assignment operator that is messing with your head.  Do you like `x == DartOptions.fromMap(x.toMap())` better?

Reply:
I think it's more than when I start reading I don't know what `x` is, and then it's on the other side too, so I'm left staring at it wondering if I'm supposed to know what `x` is from some other context or if I need to solve for `x` before finally figuring out what it's trying to say. If you really want to keep the formula version, I think it needs an introduction, like "This is the inverse of `fromMap`, such that `...`"